### PR TITLE
BE-594/domain/cosmos | cosmos/auth, cosmos/tx packages

### PR DESCRIPTION
**Pull request by deividaspetraitis**
This PR was opened as part of bigger claimbot feature to keep PRs small, focused and more manageable and adds `cosmos/auth`, `cosmos/tx` packages containing various high level helper functions and wrappers abstracting lower level details interacting with the chain.

-  `cosmos/tx` package is used directly by claimbot to [contruct](https://github.com/osmosis-labs/sqs/pull/524/files#diff-c413424bd187eb3ae47b127887855905a5f91e41dce0208b397ed14388b67381R91) and [broadcast](https://github.com/osmosis-labs/sqs/pull/524/files#diff-c413424bd187eb3ae47b127887855905a5f91e41dce0208b397ed14388b67381R102) transactions.
- `cosmos/auth` package is also used directly by claimbot to retrieve [account number and sequence](https://github.com/osmosis-labs/sqs/pull/524/files#diff-c413424bd187eb3ae47b127887855905a5f91e41dce0208b397ed14388b67381R106).

Note, this PR will be merged after #525 is merged since it depends on `delivery/http` package.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new package for querying account information in the Cosmos Auth module.
	- Added functionality for building, simulating, and sending transactions within the Cosmos SDK framework, including gas and fee calculations.
	- Implemented mock clients for testing account queries and transaction functionalities.

- **Bug Fixes**
	- Enhanced error handling in mock implementations to prevent unimplemented function calls.

- **Documentation**
	- Updated documentation to reflect new structures and methods for better clarity on usage.

### CODE REVIEW

File: domain/cosmos/tx/tx.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// Account represents the account information required for transaction building and signing.
+type Account struct {
+	Sequence      uint64 // Current sequence (nonce) of the account, used to prevent replay attacks.
+	AccountNumber uint64 // Unique identifier of the account on the blockchain.
+}
```
Comment (p0mvn) [inline domain/cosmos/tx/tx.go]: Would it be possible to import this instead?
https://github.com/cosmos/cosmos-sdk/blob/b5b1d9fe35977a97cd86c58ac82b6c9000b1d715/x/auth/types/auth.pb.go#L32-L37

File: domain/cosmos/auth/types/types.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// QueryAccountResponse encapsulates the response for an account query.
+type QueryAccountResponse struct {
+	Account Account `json:"account"`
+}
+
+// GetAccount retrieves account information for a given address.
+func (c *queryClient) GetAccount(ctx context.Context, address string) (*QueryAccountResponse, error) {
+	resp, err := http.Get(ctx, c.lcd+"/cosmos/auth/v1beta1/accounts/"+address)
```
Comment (p0mvn) [inline domain/cosmos/auth/types/types.go]: I'm also curious if we could rely more on cosmos SDK client helpers, potentially minimizing reimplementation.

Please take a look at this CosmosSender for inspiration:
https://github.com/polaris-portal/polaris/blob/ebafb78bf432380072c9f74a433cf61e7336ed83/app/gasstation/senders/cosmos.go#L46

Comment (deividaspetraitis) [reply domain/cosmos/auth/types/types.go]: Yes! Great point, thanks! I have also got rid of HTTP, and switched to gRPC, it was a bit unclear how to fill all the fields for [BaseAccount](https://github.com/cosmos/cosmos-sdk/blob/b5b1d9fe35977a97cd86c58ac82b6c9000b1d715/x/auth/types/auth.pb.go#L32-L37), leaving some fields not filled would make this client really confusing.

After the changes our local implementation is a wrapper around [QueryClient](https://github.com/osmosis-labs/sqs/blob/6c45d4ce7781a35f3b82c4a2bfb567342b0c8883/domain/cosmos/auth/types/types.go#L35), that in turn required me to pull in [encodingConfig](https://github.com/osmosis-labs/sqs/blob/6c45d4ce7781a35f3b82c4a2bfb567342b0c8883/domain/cosmos/auth/types/types.go#L15) to be able to unpack response.

Let me know if that makes sense!

File: domain/cosmos/tx/tx.go (hunk 1 above)
Comment (deividaspetraitis) [reply domain/cosmos/tx/tx.go]: Updated as per you suggestion! :pray:

**Review by p0mvn — APPROVED** [review]

**Pull request merged**